ledger/catchpointtracker_test.go

```diff
@@ -1185,104 +1189,128 @@ func TestCalculateCatchpointRounds(t *testing.T) {
 func TestCatchpointFirstStageInfoPruning(t *testing.T) {
 	partitiontest.PartitionTest(t)
 
-	// create new protocol version, which has lower lookback
-	testProtocolVersion :=
-		protocol.ConsensusVersion("test-protocol-TestFirstStageInfoPruning")
-	protoParams := config.Consensus[protocol.ConsensusCurrentVersion]
-	protoParams.CatchpointLookback = 32
-	protoParams.EnableCatchpointsWithSPContexts = true
-	config.Consensus[testProtocolVersion] = protoParams
-	defer func() {
-		delete(config.Consensus, testProtocolVersion)
-	}()
-
-	accts := []map[basics.Address]basics.AccountData{ledgertesting.RandomAccounts(20, true)}
-
-	ml := makeMockLedgerForTracker(t, false, 1, testProtocolVersion, accts)
-	defer ml.Close()
+	tempDir := t.TempDir()
+	var tests = []struct {
+		hotPath  string
+		coldPath string
+	}{
+		{"", ""},
+		{"hot", "cold"},
+	}
 
-	cfg := config.GetDefaultLocal()
-	cfg.CatchpointInterval = 4
-	cfg.CatchpointTracking = 2
-	ct := newCatchpointTracker(t, ml, cfg, ".")
-	defer ct.close()
+	for _, test := range tests {
+		var testName string
+		if test.hotPath == test.coldPath {
+			testName = "dirs=same"
+		} else {
+			testName = "dirs=different"
+		}
+		t.Run(testName, func(t *testing.T) {
+			test.hotPath = filepath.Join(tempDir, test.hotPath)
+			test.coldPath = filepath.Join(tempDir, test.coldPath)
+			for _, path := range []string{test.hotPath, test.coldPath} {
+				_, err := os.Stat(path)
+				if errors.Is(err, os.ErrNotExist) {
+					err := os.MkdirAll(path, 0777)
+					require.NoError(t, err)
+				}
+			}
 
-	temporaryDirectory := t.TempDir()
-	catchpointsDirectory := filepath.Join(temporaryDirectory, trackerdb.CatchpointDirName)
-	err := os.Mkdir(catchpointsDirectory, 0777)
-	require.NoError(t, err)
+			dataFileDirectory := filepath.Join(test.hotPath, trackerdb.CatchpointDirName)
+			err := os.Mkdir(dataFileDirectory, 0777)
+			require.NoError(t, err)
 
-	ct.dbDirectory = temporaryDirectory
-	ct.tmpDir = temporaryDirectory
+			// create new protocol version, which has lower lookback
+			testProtocolVersion :=
+				protocol.ConsensusVersion("test-protocol-TestFirstStageInfoPruning")
+			protoParams := config.Consensus[protocol.ConsensusCurrentVersion]
+			protoParams.CatchpointLookback = 32
+			protoParams.EnableCatchpointsWithSPContexts = true
+			config.Consensus[testProtocolVersion] = protoParams
+			defer func() {
+				delete(config.Consensus, testProtocolVersion)
+			}()
+
+			accts := []map[basics.Address]basics.AccountData{ledgertesting.RandomAccounts(20, true)}
+
+			ml := makeMockLedgerForTracker(t, false, 1, testProtocolVersion, accts)
+			defer ml.Close()
+
+			cfg := config.GetDefaultLocal()
+			cfg.CatchpointInterval = 4
+			cfg.CatchpointTracking = 2
+			ct := newCatchpointTrackerWithPaths(t, ml, cfg, test.hotPath, test.coldPath)
+			defer ct.close()
+
+			expectedNumEntries := protoParams.CatchpointLookback / cfg.CatchpointInterval
```

Comment:
I can follow this reading carefully, but this function might be more approachable with either a few comments explaining the flow or one general one going over how you go about testing.